exporter/stackdriverexporter/spandata.go

```diff
@@ -17,293 +17,172 @@ package stackdriverexporter
-// When resource is not nil, then its labels are attached to span attributes
-func protoSpanToOCSpanData(span *tracepb.Span, resource *resourcepb.Resource) (*trace.SpanData, error) {
-	if span == nil {
-		return nil, errNilSpan
-	}
-
-	sTracestate, err := protoTracestateToOCTracestate(span.Tracestate)
-	if err != nil {
-		return nil, err
-	}
-
-	sc := trace.SpanContext{Tracestate: sTracestate}
-	copy(sc.TraceID[:], span.TraceId)
-	copy(sc.SpanID[:], span.SpanId)
-	var parentSpanID trace.SpanID
-	copy(parentSpanID[:], span.ParentSpanId)
-	startTime, _ := ptypes.Timestamp(span.StartTime)
-	endTime, _ := ptypes.Timestamp(span.EndTime)
-	sd := &trace.SpanData{
-		SpanContext:     sc,
-		ParentSpanID:    parentSpanID,
-		StartTime:       startTime,
-		EndTime:         endTime,
-		Name:            derefTruncatableString(span.Name),
-		Attributes:      protoAttributesToOCAttributes(span.Attributes, resource),
-		Links:           protoLinksToOCLinks(span.Links),
-		Status:          protoStatusToOCStatus(span.Status),
-		SpanKind:        protoSpanKindToOCSpanKind(span.Kind),
-		MessageEvents:   protoTimeEventsToOCMessageEvents(span.TimeEvents),
-		Annotations:     protoTimeEventsToOCAnnotations(span.TimeEvents),
-		HasRemoteParent: protoSameProcessAsParentToOCHasRemoteParent(span.SameProcessAsParentSpan),
-	}
-
-	return sd, nil
-}
-
-func derefTruncatableString(ts *tracepb.TruncatableString) string {
-	if ts == nil {
-		return ""
+func pdataResourceSpansToOTSpanData(rs pdata.ResourceSpans) ([]*export.SpanData, error) {
+	resource := rs.Resource()
+	var sds []*export.SpanData
+	ilss := rs.InstrumentationLibrarySpans()
+	for i := 0; i < ilss.Len(); i++ {
+		ils := ilss.At(i)
+		spans := ils.Spans()
+		for j := 0; j < spans.Len(); j++ {
+			sd, err := pdataSpanToOTSpanData(spans.At(j), resource, ils.InstrumentationLibrary())
+			if err != nil {
+				return nil, err
+			}
+			sds = append(sds, sd)
+		}
 	}
-	return ts.Value
-}
 
-func protoStatusToOCStatus(s *tracepb.Status) (ts trace.Status) {
-	if s == nil {
-		return
-	}
-	return trace.Status{
-		Code:    s.Code,
-		Message: s.Message,
-	}
+	return sds, nil
 }
 
-func protoTracestateToOCTracestate(ts *tracepb.Span_Tracestate) (*tracestate.Tracestate, error) {
-	if ts == nil {
-		return nil, nil
+func pdataSpanToOTSpanData(
+	span pdata.Span,
+	resource pdata.Resource,
+	il pdata.InstrumentationLibrary,
+) (*export.SpanData, error) {
+	if span.IsNil() {
+		return nil, errNilSpan
 	}
-	return tracestate.New(nil, protoEntriesToOCEntries(ts.Entries)...)
-}
-
-func protoEntriesToOCEntries(protoEntries []*tracepb.Span_Tracestate_Entry) []tracestate.Entry {
-	ocEntries := make([]tracestate.Entry, 0, len(protoEntries))
-	for _, protoEntry := range protoEntries {
-		var entry tracestate.Entry
-		if protoEntry != nil {
-			entry.Key = protoEntry.Key
-			entry.Value = protoEntry.Value
+	sc := apitrace.SpanContext{}
+	copy(sc.TraceID[:], span.TraceID())
```

Comment:
Why do you have to copy slice items rather than just `sc.TraceID = span.TraceID()`?

Reply:
I see that this is how it was done before, feel fee to keep it then.